### PR TITLE
No issue: Adding new start on homepage setting UI test

### PR DESCRIPTION
**Pull request by sv-ohorvath**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Review by AndiAJ — APPROVED** [review]
👍🏽

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsHomepageTest.kt [hunk 1]
```diff
@@ -208,6 +197,29 @@ class SettingsHomepageTest {
...
+        }.goBack {}
+
+        with(activityIntentTestRule) {
+            finishActivity()
+            mDevice.waitForIdle()
+            openAppFromExternalLink(genericPage.url.toString())
+        }
+
+        browserScreen {
+            dismissCookieProtectionHint()
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsHomepageTest.kt:218]: @Mugurell Was there a way to disable TCP after activity launch, or was that previous crash still a problem? I'm just wondering as then we won't need to dismiss it here as part of a test.

Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SettingsHomepageTest.kt]: I made some changes and now I'm disabling TCP before re-launching the app. Please have another look

Comment (Mugurell) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SettingsHomepageTest.kt]: The `applySettingsExceptions` method Oana used in the new patch version is the right approach to immediately modify these settings in individual tests while also ensuring their proper cleanup.

**Review by AaronMT — APPROVED** [review]

**Pull request merged**